### PR TITLE
Feature/#10 review build system

### PR DESCRIPTION
**Pull request by fjcalzado**
The following changes have been done:

- Babel configuration `.babelrc` updated to support both provfiles `commonjs` and `es` through environment variable. 
- `es` profile fixed to target output in pure ES6 syntax.
- Added version number in UMD bundle file name. E.g.: `react-promise-tracker-0.0.2.js` and `react-promise-tracker-0.0.2.min.js`.
- Added proper library names for each module format under UMD output mode.
- Improve discoverability of the package (though Google, npm search, etc) by adding `package.json` social fields as: `keywords`, `homepage`, `contributors`, etc. Fields reordered based on NPM documentation.

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -1,6 +1,26 @@
...
+      ]
+    },
+    // Future Profile. ES6 output and module syntax.
+    "es": {
+      "presets": [
+        ["env", {
+          "useBuiltIns": false,
+          "modules": false,
+          "targets": {
+            "node": "6.5"
```
Comment (nasdan) [inline .babelrc:19]: Why did you need to fix node version?

File: .babelrc [hunk 2]
```diff
@@ -1,6 +1,26 @@
...
+          }
+        }],
+        "react"
+      ]
+    }
+  }
 }
```
Comment (nasdan) [inline .babelrc]: Care with no `end of line`. Missing add `.editorconfig` to this project?

Comment (fjcalzado) [reply .babelrc]: Yes, that is something I missed too. No linters and no `.editorconfig`. I supposed that given it is a tiny project you did not want to add extra complexity, but I will be happy to add it if you prefer so. Do you?

File: .babelrc (hunk 1 above)
Comment (fjcalzado) [reply .babelrc:19]: Yeah Dani, this is a way to make sure Babel transpiles to full ES6. Maybe is not that elegant, but the problem is `babel-preset-env` cannot be targeted to a specific ECMAScript version (e.g.: ES6). They lack a setting to do so unfortunately. Actually, they have some open tickets requesting this feature.
Instead, you can target towards a browser version or a node version. This way, the preset will configure Babel by enabling/disabling the required plugins/transformations for that particular version. 

So, in order to make sure we transpile to ES2015, we have to options:
- Manually enable plugins and transformations for ES7, ES8, etc features and disable the corresponding for ES6.
- Or, a simpler aproach, target to Node 6.5 wich was the first version with full ES6 compatibility. 

You can check it with [Babel REPL](https://babeljs.io/repl/).

If you guys know a better way to achieve this, please let me know.

File: .babelrc (hunk 2 above)
Comment (nasdan) [reply .babelrc]: Yes, you are right, it's a tiny project. In my opinion, `.editorconfig` has 0 configuration and it gives too much. Linters could be more difficult to implement because rules depends by type of project.

For me it's ok if we add `.editorconfig` as you wish

Comment (fjcalzado) [reply .babelrc]: Sure @Nasdan, I have include it. Thanks for the tip!

File: .babelrc (hunk 1 above)
Comment (fjcalzado) [reply .babelrc:19]: If you think guys this could be a problem, I can remove `es` build target, it is not mandatory in NPM packages. Whatever you prefer. This is the only thing pending to confirm, I think the rest is ready.
